### DISCUSSION TITLE
Getting the asset group for a given asset key in Python

### DISCUSSION THREAD

**Discussion by cojoko** [dagster-io/dagster#19273] [Q&A]
Hi all,
I've been trying to figure out a way to get the asset group for an asset key within an op, with no luck. For context, I am working within an op and I have access to the context and instance of the assets I am trying to get the group for, but I have not been able to find a way to see what group an asset belongs to, even if I have the full asset key. Thank you for your help!

There is a similar question which has reply suggesting adding all your asset definitions to a list and iterating, but that does not seem like a viable option when working at scale with many definitions scattered around a project.

**Comment by jamiedemaria** [comment]
I poked around the code base a bit a didn't see any methods that would map asset_key -> group for you. One option to get the list of asset definitions would be to import the `Definitions` object and then access the list there, but you might run into some scaling issues with that as well, i haven't tested 

```python
@op 
def my_op(context):
     from path.to.definitions import defs

     group_name = defs.get_assets_def("asset_key").group_name
```

**Reply by cojoko** [reply]
Thanks for taking a look! While this does work for a small deployment using a single definitions location, you are absolutely right that for a scaled production deployment which uses many definitions across many repositories you cannot point to a single location for definitions. If there was a way to retrieve all definitions for a given instance, however, it would be possible to to call `.get_assets_def` on that, but I don't think that is possible either.
